File: Codes/3541-find-most-frequent-vowel-and-consonant.java
/*
 * @Author: LetMeFly
 * @Date: 2025-09-13 16:07:42
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-09-13 16:23:21
 */
class Solution {
    private final static int mask = 1<<0 | 1<<4 | 1<<8 | 1<<14 | 1<<20;

    public int maxFreqSum(String s) {
        int[] times = new int[26];
        for (int i = 0; i < s.length(); i++) {
            times[s.charAt(i) - 'a']++;
        }
        int[] cnt = new int[2];
        for (int i = 0; i < 26; i++) {
            int idx = mask >> i & 1;
            cnt[idx] = Math.max(cnt[idx], times[i]);
        }
        return cnt[0] + cnt[1];
    }
}

File: Codes/3541-find-most-frequent-vowel-and-consonant.rs
/*
 * @Author: LetMeFly
 * @Date: 2025-09-13 16:07:42
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-09-13 16:32:24
 */
impl Solution {
    const MASK: usize = 1<<0 | 1<<4 | 1<<8 | 1<<14 | 1<<20;

    pub fn max_freq_sum(s: String) -> i32 {
        let mut times: Vec<i32> = vec![0; 26];
        for c in s.bytes() {
            times[(c - b'a') as usize] += 1;
        }
        let mut cnt: Vec<i32> = vec![0; 2];
        for i in 0..26 {
            let idx: usize = Self::MASK >> i & 1;
            cnt[idx] = cnt[idx].max(times[i]);
        }
        cnt[0] + cnt[1]
    }
}